src/freetype_manager.cc

```diff
@@ -0,0 +1,596 @@
+            gFtFontDescriptors[font].maxHeight = 0;
+            gFtFontDescriptors[font].filebuffer = NULL;
+        } else {
+            ++gFtFontsLength;
+
+            if (currentFont == -1) {
+                currentFont = font;
+            }
+
+            gFtFontManager.maxFont = gFtFontsLength + 100;
+        }
+    }
+
+    if (currentFont == -1) {
+        return -1;
+    }
+
+    gFtFontsInitialized = true;
+
+    FtFontSetCurrentImpl(currentFont + 100);
+
+    return 0;
+}
+
+// 0x441CEC
+void FtFontsExit()
+{
+    for (int font = 0; font < FT_FONT_MAX; font++) {
+        if (gFtFontDescriptors[font].filebuffer != NULL) {
+            internal_free_safe(gFtFontDescriptors[font].filebuffer, __FILE__, __LINE__); // FONTMGR.C, 124
+        }
+    }
+    //TODO: clean up
+}
+
+// 0x441D20
+static int FtFontLoad(int font_index)
+{
+    char string[56];
+    FtFontDescriptor* desc = &(gFtFontDescriptors[font_index]);
+
+    Config config;
+    if (!configInit(&config)) {
+        return -1;
+    }
+
+    sprintf(string, "fonts/%s/font.ini", settings.system.language.c_str());
+    if (!configRead(&config, string, false)) {
+        return -1;
+    }
+
+    sprintf(string, "font%d", font_index);
+
+    if (!configGetInt(&config, string, "maxHeight", &desc->maxHeight)) {
+        return -1;
+    }
+    if (!configGetInt(&config, string, "maxWidth", &desc->maxWidth)) {
+        desc->maxWidth = desc->maxHeight;
+    }
+    if (!configGetInt(&config, string, "lineSpacing", &desc->lineSpacing)) {
+        return -1;
+    }
+    if (!configGetInt(&config, string, "wordSpacing", &desc->wordSpacing)) {
+        return -1;
+    }
+    if (!configGetInt(&config, string, "letterSpacing", &desc->letterSpacing)) {
+        return -1;
+    }
+    if (!configGetInt(&config, string, "heightOffset", &desc->heightOffset)) {
+        return -1;
+    }
+    if (!configGetInt(&config, string, "warpMode", &desc->warpMode)) {
+        return -1;
+    }
+    char* encoding = NULL;
+    if (!configGetString(&config, string, "encoding", &encoding)) {
+        return -1;
+    }
+    strcpy(desc->encoding, encoding);
+    
+    char *fontFileName = NULL;
+    if (!configGetString(&config, string, "fileName", &fontFileName)) {
+        return -1;
+    }
+
+    sprintf(string, "fonts/%s/%s", settings.system.language.c_str(), fontFileName);
+
+    File* stream = fileOpen(string, "rb");
+    if (stream == NULL) {
+        return -1;
+    }
+
+    int fileSize = fileGetSize(stream); //19647736
+
+    desc->filebuffer = (unsigned char*)internal_malloc_safe(fileSize, __FILE__, __LINE__); // FONTMGR.C, 259
+
+    int readleft = fileSize;
+    unsigned char* ptr = desc->filebuffer;
+
+    while (readleft > 10000) {
+        int readsize = fileRead(ptr, 1, 10000, stream);
+        if (readsize != 10000) {
+            return -1;
+        }
+        readleft -= 10000;
+        ptr += 10000;
+    }
+
+    if (fileRead(ptr, 1, readleft, stream) != readleft) {
```

Comment:
If this fails, what happens with desc->filebuffer ? Is it cleaned up by the caller?